### ISSUE TITLE
Suggestion: Enforce 80 character lines with Eslint

### ISSUE THREAD

**Issue by SeanPrashad** [mozilla/addons-frontend#4940]
### Describe the problem and steps to reproduce it:
Whilst working on #4693, multiple reviewers mentioned that my lines exceeded 80 characters and thus were too long, however eslint did not complain. For example, [here](https://github.com/mozilla/addons-frontend/pull/4693/files#diff-7ac6d4cc535e144232c13e382e52eeecR176) is 95 characters long.

To reproduce the problem, simply type out any valid line of code that is greater than 80 characters in length.

### What happened?
`yarn lint` did not report any errors with lines > 80 characters.

### What did you expect to happen?
`yarn lint` should complain when lines are > 80 characters and the build should fail on CI.

### Anything else we should know?
It looks like the currently `.eslintrc` configuration extends from [`amo`](https://github.com/mozilla/addons-frontend/blob/master/.eslintrc#L2) - that would be a good place to begin investigations.

In addition, it might be worthwhile to use [Prettier pre-commit hooks](https://prettier.io/docs/en/precommit.html) to automatically format code when commits are made based on the existing `eslint` config.

**Comment by SeanPrashad** [comment]
[Eslint rule for `max-len`](https://eslint.org/docs/rules/max-len) to investigate

**Comment by bobsilverberg** [comment]
While we *prefer* that lines be limited to 80 chars, I think we intentionally do not have an eslint rule for that so that we can break the rule on the rare occasions where it's the only way to make the code look decent. Therefore, I'm not sure if we want to implement a real 80 char limit via eslint.

@willdurand @muffinresearch what do you think?

**Comment by bobsilverberg** [comment]
We could do that, but I personally would find that annoying if I have to ignore some warnings when running it locally. It would make quickly determining whether lint passed or not into a more difficult chore.

**Comment by willdurand** [comment]
I was thinking about this too, so let's keep things as is.

**Comment by muffinresearch** [comment]
FWIW we've covered this ground before [1] and I looked at prettier in [2]. 

In the end the decision was made to maintain the status quo via curate eslint rules. Just implementing max-len was a huge pain. 

That said, if enough of the team want to revisit Prettier then we could (I like the idea of getting rid of all stylistic bike-shedding despite the rules it gets wrong IMHO) - it would however require everyone being able to deal with some rules not following their preferences before starting down that road 😀 - see the PR [2] for the areas of the contention.

[1] https://github.com/mozilla/addons-frontend/issues/3045
[2] https://github.com/mozilla/addons-frontend/pull/3059
